src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+          paramPort = portNumber;
+        }
+
+        // Validate the MAC Address.
+        if (!validateMac(paramMac)) {
+          parse_error = true;
+          msgStr      = wolStr;
+          msgStr     += F("Error, MAC Addr Invalid [");
+          msgStr     += paramMac;
+          msgStr     += F("]");
+          addLog(LOG_LEVEL_INFO, msgStr);
+        }
+
+        // Validate IP Address.
+        if (!validateIp(paramIp)) {
+          parse_error = true;
+          msgStr      = wolStr;
+          msgStr     += F("Error, IP Addr Invalid [");
+          msgStr     += paramIp;
+          msgStr     += F("]");
+          addLog(LOG_LEVEL_INFO, msgStr);
+        }
+
+        // Validate UDP Port.
+        if (!validatePort(paramPort)) {
+          parse_error = true;
+          msgStr      = wolStr;
+          msgStr     += F("Error, Port Invalid [");
+          msgStr     += paramPort;
+          msgStr     += F("]");
+          addLog(LOG_LEVEL_INFO, msgStr);
+        }
+
+        // If no errors we can send Magic Packet.
+        if (parse_error == true) {
+          msgStr = F("CMD Syntax Error");
+          addLog(LOG_LEVEL_INFO, wolStr + msgStr);
+          msgStr += F(" <br>");
+          SendStatus(event->Source, msgStr); // Reply (echo) to sender. This will print message on browser.
+        }
+        else {                               // No parsing errors, Send Magic Packet (Wake Up the MAC).
+          msgStr  = wolStr;
+          msgStr += F("MAC= ");
+          msgStr += paramMac;
+          msgStr += F(", IP= ");
+          msgStr += paramIp;
+          msgStr += F(", Port= ");
+          msgStr += paramPort;
+          addLog(LOG_LEVEL_INFO, msgStr);
+
+          // Send Magic Packet.
+          if (WiFi.status() == WL_CONNECTED) {
+            IPAddress local_IP;
+            local_IP.fromString(paramIp);
+            WOL.setBroadcastAddress(local_IP);
+
+            // WOL.setRepeat(1, 0); // One Magic Packet, No Repeats. (Library default)
+
+            if (!WOL.sendMagicPacket(paramMac, paramPort.toInt())) {
+              msgStr  = wolStr;
+              msgStr += F("Error, Magic Packet Failed (check parameters)");
+              addLog(LOG_LEVEL_INFO, msgStr);
+            }
+          }
+          else {
+            msgStr  = wolStr;
+            msgStr += F("Error, WiFi Off-Line");
+            addLog(LOG_LEVEL_INFO, msgStr);
+          }
+        }
+      }
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      break;
+    }
+  }
+  return success;
+}
+
+// ************************************************************************************************
+// validateIp(): IPv4 Address validity checker.
+// Arg: IP Address String in dot separated format (192.168.1.255).
+// Return true if IP string appears legit.
+bool validateIp(String ipStr) {
+  IPAddress ip;
+  unsigned int length = ipStr.length();
+
+  if ((length < IP_MIN_SIZE_P101) || (length > IP_ADDR_SIZE_P101)) {
+    return false;
+  }
```

Comment:
Just curious, is it needed to check for the string length?
Is it possible the `ip.fromString` may return true on strings that do contain more or less characters?
What about IPv6?

Reply:
1. ip.fromString does not fully test the IP length, so some irregular cases can get through. Learned that during validation. So I check the length for irregular length before calling ip.fromString().
2. IPv6 is not supported by the WOL plugin.